pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -194,14 +194,28 @@ func (r *ReconcileNetworkAddonsConfig) Reconcile(request reconcile.Request) (rec
 		return reconcile.Result{}, err
 	}
 
+	openshiftNetworkConfig, prev, err := r.renderPrepare(networkAddonsConfig)
```

Comment:
This is a bad function signature, it should not return openshiftNetworkConfig. If you need it for renderObjects, move out the function that fetches the openshiftNetworkConfig on this level and pass it to renderPrepare. Also, why does renderPrepare returns only `prev`. Please don't create misc functions that do everything and return random stuff. Keep it logical